sdk/src/main/java/io/opentdf/platform/sdk/Autoconfigure.java

```diff
@@ -0,0 +1,749 @@
+            this.splitID = splitId;
+        }
+
+        public boolean equals(Object obj) {
+            if (this == obj) {
+                return true;
+            }
+            if (obj == null || !(obj instanceof SplitStep)) {
+                return false;
+            }
+            SplitStep ss = (SplitStep) obj;
+            if ((this.kas.equals(ss.kas)) && (this.splitID.equals(ss.splitID))){
+                return true;
+            }
+            return false;
+        }
+    }
+
+    // Utility class for an attribute name FQN.
+    public static class AttributeNameFQN {
+        private final String url;
+        private final String key;
+
+        public AttributeNameFQN(String url) throws AutoConfigureException {
+            Pattern pattern = Pattern.compile("^(https?://[\\w./-]+)/attr/([^/\\s]*)$");
+            Matcher matcher = pattern.matcher(url);
+            if (!matcher.find() || matcher.group(1) == null || matcher.group(2) == null) {
+                throw new AutoConfigureException("invalid type: attribute regex fail");
+            }
+
+            try {
+                URLDecoder.decode(matcher.group(2), StandardCharsets.UTF_8.name());
+            } catch (Exception e) {
+                throw new AutoConfigureException("invalid type: error in attribute name [" + matcher.group(2) + "]");
+            }
+
+            this.url = url;
+            this.key = url.toLowerCase();
+        }
+
+        @Override
+        public String toString() {
+            return url;
+        }
+
+        public AttributeValueFQN select(String value) throws AutoConfigureException {
+            String newUrl = String.format("%s/value/%s", url, URLEncoder.encode(value, StandardCharsets.UTF_8));
+            return new AttributeValueFQN(newUrl);
+        }
+
+        public String prefix() {
+            return url;
+        }
+
+        public String getKey() {
+            return key;
+        }
+
+        public String authority() throws AutoConfigureException {
+            Pattern pattern = Pattern.compile("^(https?://[\\w./-]+)/attr/[^/\\s]*$");
+            Matcher matcher = pattern.matcher(url);
+            if (!matcher.find()) {
+                throw new AutoConfigureException("invalid type");
+            }
+            return matcher.group(1);
+        }
+
+        public String name() throws AutoConfigureException {
+            Pattern pattern = Pattern.compile("^https?://[\\w./-]+/attr/([^/\\s]*)$");
+            Matcher matcher = pattern.matcher(url);
+            if (!matcher.find()) {
+                throw new AutoConfigureException("invalid attribute");
+            }
+            try {
+                return URLDecoder.decode(matcher.group(1), StandardCharsets.UTF_8.name());
+            } catch (Exception e) {
+                throw new AutoConfigureException("invalid type");
+            }
+        }
+    }
+
+    // Utility class for an attribute value FQN.
+    public static class AttributeValueFQN {
```

Comment:
You may want to make these public static classes top level classes (in new files)